docs/rfcs/0002-session-id-sharing.md

```diff
@@ -0,0 +1,226 @@
+- Status:
+- Date: 2023-06-07
+- Author: [@brizental](https://github.com/brizental)
+- RFC PR: [#7112](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/7112)
+- Implementation GitHub issue: TODO
+
+## Problem Statement
+
+The VPN application is divided in two processes: the main application and the daemon. The main application is where the user interface runs and the daemon is what manages the VPN configuration and tunnel.
+
+In desktop environments, even while backgrounded the main application process is still running, but in mobile environments it is not. Due to this caveat, the mobile daemon processes have a separate instance of Glean to be able to collect telemetry about the connection while the application is in the background.
+
+This architecture poses a problem for analyzers. It is not possible to easily join the data from the same session between daemon and main application. The session ids are different due to these being two different Glean instances.
+
+For more information on this topic, refer to [Mozilla VPN Telemetry Refactor](https://docs.google.com/document/d/1jyNZ_g_cUpZZsEr2hYwnwZgkxlvqTmoFUJKp_LAOPts/edit#heading=h.xararkvsnpss).
+
+## Proposed Solution
+
+> **Note** Honestly, I think for this proposed solution it's just easier to look at the pseudo-code in Appendix 1.
+
+A new metric will be added to both the daemonsession and the session pings: shared_session_id. This metric will have lifetime ping i.e. it is cleared from storage whenever a ping containing it is submitted. Thus it needs to be set again every time a new session/daemonsession ping is submitted.
```

Comment:
The design doesn't really require it to be use lifetime.

Reply:
If we're having to manually set it before every ping - and often to the same thing it was already set to - that seems like a candidate for `user` lifetime.

I don't feel super-strongly about this, but it's how we handle `session_id`. It may make sense to align these two.

Reply:
And thinking a bit more, is this metric taking over from `session_id`? Will we deprecate `session_id` after this?

Reply:
Nope, this is a different metrics `shared_session_id`. After it's implemented and validated we can discuss maybe merging the two, but for now I'd rather have them separate.

Reply:
I have rewritten the whole proposal and the rewrite has this metrics as `user` lifetime as suggested. Resolving this.